src/main/java/de/zalando/aruha/nakadi/repository/db/SubscriptionDbRepository.java

```diff
@@ -0,0 +1,82 @@
+package de.zalando.aruha.nakadi.repository.db;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.zalando.aruha.nakadi.domain.Subscription;
+import de.zalando.aruha.nakadi.exceptions.DuplicatedSubscriptionException;
+import de.zalando.aruha.nakadi.exceptions.InternalNakadiException;
+import de.zalando.aruha.nakadi.exceptions.NoSuchSubscriptionException;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Set;
+
+import static com.google.common.collect.Sets.newTreeSet;
+
+public class SubscriptionDbRepository extends AbstractDbRepository {
+
+    private final SubscriptionMapper rowMapper = new SubscriptionMapper();
+
+    public SubscriptionDbRepository(final JdbcTemplate jdbcTemplate, final ObjectMapper objectMapper) {
+        super(jdbcTemplate, objectMapper);
+    }
+
+    public void saveSubscription(final Subscription subscription) throws InternalNakadiException,
```

Comment:
@v-stepanov Vhtang! Ktulhu is coming.

On this stage there are 2 possible ways to get problems!
 - Duplicate primary key
 - Unique constraint for app, event_types and use_case

And this method is trying to mix several different actions into one.
According to use cases that we discussed a lot, there are basically 2 methods related to 'work with stored'/'storing' subscription:
 - `Create or get subscription` based on (app, use_case, event_types)
 - `get subscription` by nakadi generated id (won't discuss the format)
These methods may be directly mapped to dao layer (as you see - they are even called like dao methods)
According to this PR, we actually need to add another method - get subscription by app_id, et and so on. 
I wold prefer another method (if we really need this nakadi id generator):

```
storedSubscription = saveSubscription(subscription, idGenerator); // idGenerator = Supplier<String>
```